firebase-firestore/src/androidTest/java/com/google/firebase/firestore/WriteBatchTest.java

```diff
@@ -222,7 +222,7 @@ public void testWriteTheSameServerTimestampAcrossWrites() {
     assertTrue(localSnap.getMetadata().hasPendingWrites());
     assertEquals(asList(map("when", null), map("when", null)), querySnapshotToValues(localSnap));
 
-    QuerySnapshot serverSnap = accumulator.await();
+    QuerySnapshot serverSnap = accumulator.awaitRemoteEvent();
```

Comment:
This PR somehow changed the timing of events so much as the `fromCache:false` event is now delivered before `hasPendingWrites:false`. I am not 100% convinced that this is due to a bug, but if I look at the test logs everything looks sane.